library/java/net/openid/appauth/JsonUtil.java

```diff
@@ -253,7 +253,9 @@ public static Long getLongIfDefined(
             return null;
         }
 
-        return json.getLong(field);
+        long candidate = json.optLong(field);
+
+        return candidate != 0 ? candidate : null;
```

Comment:
Am I right that this might introduce a bug, where the number is _explicitly_ zero in the JSON and this function would then return null? I think you should revert this change if so.

Reply:
The defintion of `optLong` is this:

```java
public long optLong(String name) {
    return optLong(name, 0L);
}
```

This is because `long` is a primitive and cannot be represented as null. Would you like me to try an implementation that uses `Long` instead?